File: app/src/main/java/com/example/designpaternpractise/behavioral/memento/java/test/CareTaker.java
package com.example.designpaternpractise.behavioral.memento.java.test;

import java.util.ArrayList;
import java.util.List;

public class CareTaker {


    private List<Memento> mementoList = new ArrayList<>();


    public List<Memento> getMementoList() {
        return mementoList;
    }

    public void addToList(Memento memento) {
        mementoList.add(memento);
    }
}

File: app/src/main/java/com/example/designpaternpractise/behavioral/memento/java/kotlin/CareTaker.kt
package com.example.tutorialdp.behaviorals.memento.java.kotlin

class CareTaker {

    private val mementoList: ArrayList<Memento> = arrayListOf()

    fun add(memento: Memento) {
        mementoList.add(memento)
    }

    fun get(index: Int): Memento? {
        if (index < mementoList.size) {
            return mementoList[index]
        }
        return null
    }
}